### PR TITLE
Fix: handle subpackage tags in is_backport

### PR DESCRIPTION
**Pull request by IanButterworth**
Fixes https://github.com/JuliaRegistries/TagBot/issues/343

Needs tests.

### CODE REVIEW

**Comment by IanButterworth** [comment]
I think this needs to only regard old tags for the same package though.. so might need to be smarter..

**Comment by IanButterworth** [comment]
@tpoisot @cadojo can you check that this works on a monorepo package by changing the action to
```
      - uses: JuliaRegistries/TagBot@ib/is_backport_monorepo
 ```

**Comment by tpoisot** [comment]
will do - later tonight or monday

**Comment by tpoisot** [comment]
Still not working - the first letter is chomped, and the action crashes because the output isn't a valid version string

**Comment by IanButterworth** [comment]
Do you have a link to a run?

**Comment by IanButterworth** [comment]
If you're referring to https://github.com/PoisotLab/SpeciesDistributionToolkit.jl/actions/runs/10856429667/job/30131050466 that appears to be using the latest release not this branch

**Comment by tpoisot** [comment]
That's the one, but the entry for the subdir I was attempting to release is definitely using the correct branch - I just checked the log

**Comment by IanButterworth** [comment]
Ah I forgot that you can't just try branches on this action, it has to be built. 

I guess we just go for it and release this. The new function has a fail safe anyway. 

Can you just do a code review and I'll go ahead. Thanks.

File: tagbot/action/changelog.py
Comment (cadojo) [inline tagbot/action/changelog.py:1]: I'll run the CI once this merges. Thank you so much for the fast fix! My only comment is to remember to remove the `_is_backport:` from the logging statement.

**Review by cadojo — APPROVED** [review]

File: tagbot/action/changelog.py [hunk 1]
```diff
@@ -69,19 +69,48 @@ def _previous_release(self, version_tag: str) -> Optional[GitRelease]:
...
+
+                # Check if the version is a backport
+                if tag_ver > cur_ver:
+                    return True
+
+            return False
+        except Exception as e:
+            # This is a best-effort function so we don't fail the entire process
+            logger.error(
+                f"An error occurred while check if the tag is a backport. Assuming False: {e}"
```
Comment (IanButterworth) [inline tagbot/action/changelog.py:111]:
```suggestion
                f"Checking if this is a backport failed. Assuming False: {e}"
```

File: tagbot/action/changelog.py [hunk 2]
```diff
@@ -69,19 +69,48 @@ def _previous_release(self, version_tag: str) -> Optional[GitRelease]:
...
+                # Check if the version is a backport
+                if tag_ver > cur_ver:
+                    return True
+
+            return False
+        except Exception as e:
+            # This is a best-effort function so we don't fail the entire process
+            logger.error(
+                f"Checking if this is a backport failed. Assuming False: {e}"
+            )
```
Comment (IanButterworth) [inline tagbot/action/changelog.py:112]:
```suggestion
            logger.error(f"Checking if backport failed. Assuming False: {e}")
```

**Pull request merged**